src/BinaryParsers/PEBinary/ProgramDatabase/LinkerCommandLine.cs

```diff
@@ -100,5 +100,44 @@ public LinkerCommandLine(string commandLine)
                 this.IncrementalLinking = false;
             }
         }
+
+        public static bool IsLinkerCommandLine(string commandLine)
+        {
+            // The command line for link.exe should contain /OUT with the final binary name.  Use that to separate compiler and linker command lines.
+            // We should accept both / and - as the argument prefix.  The argument is case insensitive.  We do not know where in the command-line the
+            // OUT argument will be provided so we use Contains.
+            return ((commandLine != null) &&
+                    (commandLine.Contains("/OUT", System.StringComparison.OrdinalIgnoreCase) || commandLine.Contains("-OUT", System.StringComparison.OrdinalIgnoreCase)));
+        }
+
+        private static bool ArgumentStartsWith(string target, string argument)
+        {
+            // Arguments are expected to begin with a - or a /.  Accept either.  They are not case sensitive.
+            if (target.Length == 0)
+            {
+                return false;
+            }
+            else if ((target[0] != '/') && (target[0] != '-'))
+            {
+                return false;
+            }
+
+            return target.Substring(1).StartsWith(argument, System.StringComparison.OrdinalIgnoreCase);
+        }
+
+        private static bool ArgumentEquals(string target, string argument)
+        {
+            // Arguments are expected to begin with a - or a /.  Accept either.  They are not case sensitive.
+            if (target.Length == 0)
+            {
+                return false;
+            }
+            else if ((target[0] != '/') && (target[0] != '-'))
+            {
+                return false;
+            }
+
+            return target.Substring(1).Equals(argument, System.StringComparison.OrdinalIgnoreCase);
```

Comment:
>target [](http://example.com/codeflow?start=19&length=6)

we could use a span here, yes? and prevent the garbage string creation.

Reply:
Unsure.  I don't spend much time in C# so I'll have to read up on that.

Reply:
Thanks @asklar for the offline pointer to some helpful docs on that.  Fortunately that was an easy fix.

Reply:
Excellent, thanks.